From 2963c8775e91ee0cea9fd21dbcb92ac0476dc887 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Fri, 1 Feb 2019 13:29:37 +0530
Subject: [PATCH] macos: lipo outputs aarch64 as arm64, fix it

Or we'll reject arm64 objects as invalid when building for aarch64
---
 mesonbuild/mesonlib.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mesonbuild/mesonlib.py b/mesonbuild/mesonlib.py
index 939b4cb1..540fcdca 100644
--- a/mesonbuild/mesonlib.py
+++ b/mesonbuild/mesonlib.py
@@ -475,6 +475,7 @@ def darwin_get_object_archs(objpath):
     stdo = stdo.rsplit(': ', 1)[1]
     # Convert from lipo-style archs to meson-style CPUs
     stdo = stdo.replace('i386', 'x86')
+    stdo = stdo.replace('arm64', 'aarch64')
     # Add generic name for armv7 and armv7s
     if 'armv7' in stdo:
         stdo += ' arm'
-- 
2.20.1

